packages/react/src/components/DataTable/Table.tsx

```diff
@@ -5,23 +5,57 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import React, { useContext } from 'react';
+import React, { useContext, PropsWithChildren } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 import { usePrefix } from '../../internal/usePrefix';
 import { TableContext } from './TableContext';
 
+interface TableProps {
+  className: string;
+
+  /**
+   * `false` If true, will apply sorting styles
+   */
+  isSortable: boolean;
+
+  /**
+   * Specify whether the overflow menu (if it exists) should be shown always, or only on hover
+   */
+  overflowMenuOnHover: boolean;
+
+  /**
+   *  Change the row height of table. Currently supports `xs`, `sm`, `md`, `lg`, and `xl`.
+   */
+  size: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
+
+  /**
+   * `false` If true, will keep the header sticky (only data rows will scroll)
+   */
+  stickyHeader: boolean;
+
+  /**
+   * `false` If true, will use a width of 'auto' instead of 100%
+   */
+  useStaticWidth: boolean;
+
+  /**
+   * `true` to add useZebraStyles striping.
+   */
+  useZebraStyles: boolean;
+}
+
```

Comment:
should these all be optional?
```suggestion
interface TableProps {
  className?: string;

  /**
   * `false` If true, will apply sorting styles
   */
  isSortable?: boolean;

  /**
   * Specify whether the overflow menu (if it exists) should be shown always, or only on hover
   */
  overflowMenuOnHover?: boolean;

  /**
   *  Change the row height of table. Currently supports `xs`, `sm`, `md`, `lg`, and `xl`.
   */
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';

  /**
   * `false` If true, will keep the header sticky (only data rows will scroll)
   */
  stickyHeader?: boolean;

  /**
   * `false` If true, will use a width of 'auto' instead of 100%
   */
  useStaticWidth?: boolean;

  /**
   * `true` to add useZebraStyles striping.
   */
  useZebraStyles?: boolean;
}

```

Reply:
@francinelucca but since `size` will be option is there a value to set as default size.